File: Basic Array Questions/sort 0,1,2 in 1D array/sort_0_1_2.cpp
// test case:
//  0 0 2 1 0 2 0 1 2 0 0 0 1 2 2 1 1 0 1 2 0 1 2 0 0 1 1 1

#include<iostream>
#include<vector>
using namespace std;

void printArray(vector<int>arr){
     for(int i=0;i<arr.size();i++){
          cout<<arr[i]<<" ";
     }
     cout<<endl;
}

int main(){
     vector<int>arr{0,0,2,1,0,2,0,1,2,0,0,0,1,2,2,1,1,0,1,2,0,1,2,0,1,1,1};
     int low=0;
     int mid=0;
     int high=arr.size()-1;
     while(mid<=high){
          if(arr[mid]==0){
               int d=arr[low];
               arr[low]=arr[mid];
               arr[mid]=d;
               mid++;
               low++;
          }
          else if(arr[mid]==1){
               mid++;
          }
          else if(arr[mid]==2){
               int d=arr[mid];
               arr[mid]=arr[high];
               arr[high]=d;
               high--;
          }
     }

     printArray(arr);
     return 0;
}

File: Basic Array Questions/sort 0,1,2 in 1D array/sort_0_1_2.py
# test case:
#   0 0 2 1 0 2 0 1 2 0 0 0 1 2 2 1 1 0 1 2 0 1 2 0 0 1 1 1

arr=[0,0,2,1,0,2,0,1,2,0,0,0,1,2,2,1,1,0,1,2,0,1,2,0,1,1,1]
low=0
mid=0
high=len(arr)-1

while(mid<=high):
     if(arr[mid]==0):
          arr[low],arr[mid]=arr[mid],arr[low]
          mid+=1
          low+=1
     elif(arr[mid]==1):
          mid+=1
     elif(arr[mid]==2):
          arr[high],arr[mid]=arr[mid],arr[high]
          high-=1
print(arr)